clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1277,10 +1284,12 @@ private void releaseAssignmentAndLeaveGroup(final Timer timer) {
         UnsubscribeEvent unsubscribeEvent = new UnsubscribeEvent(calculateDeadlineMs(timer));
         applicationEventHandler.add(unsubscribeEvent);
         try {
-            // If users subscribe to an invalid topic name, they will get InvalidTopicException in error events,
+            // If users subscribe to an invalid topic name or subscribe an authorization topic,
+            // they will get InvalidTopicException or TopicAuthorizationException in error events,
             // because network thread keeps trying to send MetadataRequest in the background.
             // Ignore it to avoid unsubscribe failed.
-            processBackgroundEvents(unsubscribeEvent.future(), timer, e -> e instanceof InvalidTopicException);
+            processBackgroundEvents(unsubscribeEvent.future(), timer,
+                    e -> e instanceof InvalidTopicException || e instanceof TopicAuthorizationException || e instanceof GroupAuthorizationException);
```

Comment:
For readability, could you introduce a `Predicate` variable, a la:

```suggestion
            final Predicate<Exception> ignoreExceptions = e ->
                e instanceof InvalidTopicException ||
                e instanceof TopicAuthorizationException ||
                e instanceof GroupAuthorizationException;
            processBackgroundEvents(unsubscribeEvent.future(), timer, ignoreExceptions);
```

Reply:
I think we need a similar fix for `AsyncKafkaConsumer#unsubscribe` as well.

Reply:
uhm are we sure that swallowing TopicAuth and GroupAuth on close is the right thing to do? I could surely be missing something, but I believe it's not what the classic consumer does, see my comment on it on the other PR that is also attempting this https://github.com/apache/kafka/pull/17516#discussion_r1824955585 
Thoughts?

Reply:
uhm are we sure that swallowing TopicAuth and GroupAuth on close is the right thing to do? I could surely be missing something, but I believe it's not what the classic consumer does, see my comment on it on the other PR that is also attempting this https://github.com/apache/kafka/pull/17516#discussion_r1824955585
Thoughts?